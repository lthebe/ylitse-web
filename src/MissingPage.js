import React from 'react';
import PropTypes from 'prop-types';
import Typography from 'material-ui/Typography';

import Page from './Page';

const MissingPage = ({ location }) => (
    <Page>
        <Typography variant="display4" gutterBottom>
            Could not find
        </Typography>
        <Typography variant="display3" gutterBottom>
            <code>{location.pathname}</code>
        </Typography>
    </Page>
);

MissingPage.propTypes = {
    location: PropTypes.shape({ pathname: PropTypes.string }).isRequired,
};

export default MissingPage;
