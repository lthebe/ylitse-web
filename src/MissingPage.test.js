import React from 'react';
import { shallow } from 'enzyme';
import MissingPage from './MissingPage';

describe('MissingPage', () => {
    test('renders correctly', () => {
        const props = { location: { pathname: '/test' } };
        const wrapper = shallow(<MissingPage {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
