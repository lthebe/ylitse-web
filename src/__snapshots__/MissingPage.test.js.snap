// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissingPage renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MissingPage
    location={
      Object {
        "pathname": "/test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          gutterBottom={true}
          variant="display3"
        >
          Could not find page:
        </WithStyles(Typography)>,
        <WithStyles(Typography)
          gutterBottom={true}
          variant="display2"
        >
          <code>
            /test
          </code>
        </WithStyles(Typography)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Could not find page:",
          "gutterBottom": true,
          "variant": "display3",
        },
        "ref": null,
        "rendered": "Could not find page:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <code>
            /test
          </code>,
          "gutterBottom": true,
          "variant": "display2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "/test",
          },
          "ref": null,
          "rendered": "/test",
          "type": "code",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            gutterBottom={true}
            variant="display3"
          >
            Could not find page:
          </WithStyles(Typography)>,
          <WithStyles(Typography)
            gutterBottom={true}
            variant="display2"
          >
            <code>
              /test
            </code>
          </WithStyles(Typography)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Could not find page:",
            "gutterBottom": true,
            "variant": "display3",
          },
          "ref": null,
          "rendered": "Could not find page:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <code>
              /test
            </code>,
            "gutterBottom": true,
            "variant": "display2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "/test",
            },
            "ref": null,
            "rendered": "/test",
            "type": "code",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
